lib/contact_screen.dart

```diff
@@ -0,0 +1,317 @@
+                      ),
+                    );
+                  },
+                );
+              },
+            ),
+    );
+  }
+
+  /// App bar for show only title and icon
+  AppBar appBarForShowOnlyTitle() {
+    return AppBar(
+      centerTitle: true,
+      automaticallyImplyLeading: false,
+      leading: Container(
+          margin: const EdgeInsets.only(left: 8),
+          child: IconButton(
+            icon: Icon(Icons.arrow_back_ios,
+                color: Theme.of(context).textSelectionTheme.selectionColor),
+            onPressed: () {
+              Navigator.pop(context);
+            },
+          )),
+      title: titleWidget(),
+      actions: [
+        IconButton(
+            icon: Icon(Icons.search_rounded,
+                color: Theme.of(context).textSelectionTheme.selectionColor),
+            onPressed: () {
+              _valueNotifier.value = !_valueNotifier.value;
+            })
+      ],
+    );
+  }
+
+  /// AppBar for Search bhajan
+  AppBar appBarForSearch() {
+    return AppBar(
+        centerTitle: true,
+        automaticallyImplyLeading: false,
+        leading: Container(
+            margin: const EdgeInsets.only(left: 8),
+            child: IconButton(
+              icon: Icon(Icons.arrow_back_ios,
+                  color: Theme.of(context).textSelectionTheme.selectionColor),
+              onPressed: () {
+                _valueNotifier.value = !_valueNotifier.value;
+              },
+            )),
+        title: titleSearchWidget(),
+        actions: <Widget>[
+          IconButton(
+            icon: Icon(Icons.close,
+                color: Theme.of(context).textSelectionTheme.selectionColor),
+            onPressed: () {
+              _valueNotifier.value = !_valueNotifier.value;
+            },
+          ),
+        ]);
+  }
+
+  void searchFromContactList() {
+    if (_searchController.text.isNotEmpty) {
+      if (_filterOn == "name") {
+        final list = _contactsList.where((element) {
+          return element.name
+              .toLowerCase()
+              .contains(_searchController.text.toLowerCase());
+        }).toList();
+        _searchNotifier.updateSearchValue(list);
+      } else if (_filterOn == "phoneNumber") {
+        final list = _contactsList.where((element) {
+          if (element.phoneNumber.isNotEmpty) {
+            return element.phoneNumber
+                .toLowerCase()
+                .contains(_searchController.text.toLowerCase());
+          } else {
+            return false;
+          }
+        }).toList();
+        _searchNotifier.updateSearchValue(list);
+      } else if (_filterOn == "kilvishId") {
+        final list = _contactsList.where((element) {
+          if (element.kilvishId != null) {
+            return element.kilvishId!
+                .toLowerCase()
+                .contains(_searchController.text.toLowerCase());
+          } else {
+            return false;
+          }
+        }).toList();
```

Comment:
The filtering logic is incorrect .. we need to filter in those contacts that (partial) match the text typed on name, phone or kilvish id. Something like
```
_contactList.forEach( element => {(
  if ( element.name.match(searchText) or element.phone.match(searchText) or element.kilvisId.match(searchText) ). {
    include the element
  }
  else {
   exclude the element
  }
})
```